pandas/core/arrays/_mixins.py

```diff
@@ -214,9 +215,21 @@ def __setitem__(self, key, value):
     def _validate_setitem_value(self, value):
         return value
 
+    @overload
+    # error: Overloaded function signatures 1 and 2 overlap with incompatible
+    # return types  [misc]
+    def __getitem__(self, key: int) -> EAScalarOrMissing:  # type: ignore[misc]
```

Comment:
this can also return NDArrayBackedExtensionArrayT

Reply:
can you elaborate. is this for 2d EA?

Reply:
NDArrayBackedExtensionArray supports 2D, exactly

Reply:
so it also supports a tuple indexer? and I guess from that NDArrayBackedExtensionArray can't support nested data as it uses is_scalar checks.

so also need to change `EAScalarOrMissing = object  # both scalar value and na_value can be any type` -> `ScalarOrScalarMissing = Scalar  # both values and na_value must be scalars` ?

Reply:
> so it also supports a tuple indexer?

Yes.  For that matter i think even 1D will support 1-tuples

> and I guess from that NDArrayBackedExtensionArray can't support nested data as it uses is_scalar checks.

PandasArray can have object dtype, and Categorical can hold tuples.